systems/controllers/QPCommon.cpp

```diff
@@ -534,26 +551,28 @@ int setupAndSolveQP(NewQPControllerData *pdata, std::shared_ptr<drake::lcmt_qp_c
     Aeq.block(6,0,neps,nq) = Jp;
     Aeq.block(6,nq,neps,nf) = MatrixXd::Zero(neps,nf);  // note: obvious sparsity here
     Aeq.block(6,nq+nf,neps,neps) = MatrixXd::Identity(neps,neps);             // note: obvious sparsity here
-    beq.segment(6,neps) = (-Jpdot -params->Kp_accel*Jp)*robot_state.qd; 
+    beq.segment(6,neps) = -Jpdotv -params->Kp_accel*Jp*robot_state.qd; 
   }    
   
   // add in body spatial equality constraints
   // VectorXd body_vdot;
-  MatrixXd orig = MatrixXd::Zero(4,1);
-  orig(3,0) = 1;
+  Vector4d orig1 = Vector4d::Zero();
+  orig1(3) = 1.0;
+  Vector3d orig = Vector3d::Zero();
   int equality_ind = 6+neps;
   MatrixXd Jb(6,nq);
-  MatrixXd Jbdot(6,nq);
+  Vector6d Jbdotv;	
   for (int i=0; i<desired_body_accelerations.size(); i++) {
     if (desired_body_accelerations[i].weight < 0) { // negative implies constraint
       if (!inSupport(active_supports,desired_body_accelerations[i].body_id0)) {
-        pdata->r->forwardJac(desired_body_accelerations[i].body_id0,orig,1,Jb);
-        pdata->r->forwardJacDot(desired_body_accelerations[i].body_id0,orig,1,Jbdot);
+        pdata->r->forwardJac(desired_body_accelerations[i].body_id0,orig1,1,Jb);
+        auto Jdot_times_v = pdata->r->forwardJacDotTimesV(orig,desired_body_accelerations[i].body_id0,0,1,0);
+        Jbdotv = Jdot_times_v.value();
```

Comment:
For future reference, this performs an unnecessary copy from `Jdot_times_v.value()` to `Jbdotv`. You can get rid of the copy by making Jbdotv a reference to the appropriate type (either auto& or I suppose Vector6d&). Not essential to fix this right now but just so you know.